__tests__/components/Form.spec.js

```diff
@@ -0,0 +1,153 @@
+  type: 'object',
+  properties: {
+    Name: {
+      title: 'Pokemon Name',
+      type: 'string'
+    }
+  }
+}
+
+describe('Form component', () => {
+  it('should match the stored snapshot', () => {
+    const component = renderer.create(
+      <Provider>
+        <Form schema={schema} />
+      </Provider>
+    )
+    let tree = component.toJSON()
+    expect(tree).toMatchSnapshot()
+  })
+
+  describe('hideSubmitButton property', () => {
+    it('should render a submit button if the property is not specified', () => {
+      const component = mount(
+        <Provider>
+          <Form schema={schema} />
+        </Provider>
+      )
+
+      expect(component.find('button')).toHaveLength(1)
+
+      component.unmount()
+    })
+
+    it('should render a submit button if the property is falsey', () => {
+      const component = mount(
+        <Provider>
+          <Form schema={schema} hideSubmitButton={false} />
+        </Provider>
+      )
+
+      expect(component.find('button')).toHaveLength(1)
+
+      component.unmount()
+    })
+
+    it('should not render a submit button if the property is truthey', () => {
+      const component = mount(
+        <Provider>
+          <Form schema={schema} hideSubmitButton />
+        </Provider>
+      )
+
+      expect(component.find('button')).toHaveLength(0)
+
+      component.unmount()
+    })
+  })
+
+  describe('submitButtonText property', () => {
+    it('should change the text value of the submit button', () => {
+      const submitText = 'Click me to submit this form'
+      const component = mount(
+        <Provider>
+          <Form schema={schema} submitButtonText={submitText} />
+        </Provider>
+      )
+
+      expect(component.find('button').text()).toEqual(submitText)
+
+      component.unmount()
+    })
+  })
+
+  describe('onSubmit property', () => {
+    it('should be called when the submit button is clicked', (done) => {
+      const value = 'Squirtle'
+      const callback = sinon.spy()
+      const component = mount(
+        <Provider>
+          <Form schema={schema} onSubmit={callback} />
+        </Provider>
+      )
+
+      const input = component.find('input')
+      input.simulate('change', { target: { value } })
+      component.update()
+
+      setTimeout(() => {
+        component.find('form').simulate('submit')
+        expect(callback.callCount).toEqual(1)
+        expect(callback.getCall(0).args[0].formData).toEqual({ Name: value })
+        done()
+      }, 150)
+    })
+  })
+
+  describe('onChange property', () => {
+    it('should be called when an input field is changed', (done) => {
+      const value = 'Squirtle'
+      const callback = sinon.spy()
+      const component = mount(
+        <Provider>
+          <Form schema={schema} onChange={callback} />
+        </Provider>
+      )
+
+      const input = component.find('input')
+      input.simulate('change', { target: { value } })
+      component.update()
+
+      setTimeout(() => {
```

Comment:
If-you-are-in-the-mood: maybe we could use a since `waitForComponentUpdate` (or something) that could
* call `component.update()`
* call `setTimeout` and return a promise (or use Bluebird's `Promise.delay()`)

Additionally this could let us drop the `done()` calls (since we will be returning the promise instead form within the `it()`) and maybe in the future even let use `await` :)